### PR TITLE
[FLINK-6766] Update documentation about async backends and incrementa…

### PR DESCRIPTION
**Pull request by StefanRRichter**
…l checkpoints

This PR introduces some documentation about async heap backends and incremental snapshots with RocksDB.

### CODE REVIEW

**Comment by StefanRRichter** [comment]
CC @rmetzger

File: docs/monitoring/large_state_tuning.md [hunk 1]
```diff
@@ -138,6 +138,22 @@ Unfortunately, RocksDB's performance can vary with configuration, and there is l
 RocksDB properly. For example, the default configuration is tailored towards SSDs and performs suboptimal
 on spinning disks.
 
+**Incremental Checkpoints**
+
+Incremental checkpoints can dramatically reduce the checkpointing time in comparison to full checkpoints, at the cost of a (potentially) longer
+recovery time. The core idea is that incremental checkpoints only record all changes to the previous completed checkpoint, instead of
+producing a full, self-contained backups of the backend. Like this, incremental checkpoints build upon previous checkpoints. Flink leverages
```
Comment (alpinegizmo) [inline docs/monitoring/large_state_tuning.md]: a full, self-contained backup of the state backend

File: docs/ops/state_backends.md [hunk 2]
```diff
@@ -88,6 +102,13 @@ that is (per default) stored in the TaskManager data directories. Upon checkpoin
 RocksDB data base will be checkpointed into the configured file system and directory. Minimal
 metadata is stored in the JobManager's memory (or, in high-availability mode, in the metadata checkpoint).
 
+The RocksDBStateBackend always performs asynchronous snapshots.
+
+Limitations of the RocksDBStateBackend:
+
+  - As RocksDB's JNI bridge API is based on byte[], the maximum supported size per key and per value is 2^31 bytes each. 
+  IMPORTANT: states that use merge operations in RocksDB (e.g. ListState) can silently accumulate value sizes > 2^31 bytes and will then fail on their next retrival. This is currently a limitation of RocksDB JNI.
+
```
Comment (alpinegizmo) [inline docs/ops/state_backends.md]: retrieval

File: docs/ops/state_backends.md [hunk 3]
```diff
@@ -88,6 +102,13 @@ that is (per default) stored in the TaskManager data directories. Upon checkpoin
 RocksDB data base will be checkpointed into the configured file system and directory. Minimal
 metadata is stored in the JobManager's memory (or, in high-availability mode, in the metadata checkpoint).
 
+The RocksDBStateBackend always performs asynchronous snapshots.
+
+Limitations of the RocksDBStateBackend:
+
+  - As RocksDB's JNI bridge API is based on byte[], the maximum supported size per key and per value is 2^31 bytes each. 
+  IMPORTANT: states that use merge operations in RocksDB (e.g. ListState) can silently accumulate value sizes > 2^31 bytes and will then fail on their next retrival. This is currently a limitation of RocksDB JNI.
```
Comment (alpinegizmo) [inline docs/ops/state_backends.md]: retrieval

**Comment by StefanRRichter** [comment]
Thanks for the input @alpinegizmo ! I updated the text accordingly.

**Comment by rmetzger** [comment]
I'll merge the change.

**Pull request closed without merging**